### DISCUSSION TITLE
When can we use latest .NET runtime for source generator?

### DISCUSSION THREAD

**Discussion by acaly** [dotnet/roslyn#49320] [Q&A] [answered]
I think it's very possible that the generator needs to use some other projects in the solution. If it is limited to net standard, the referenced project needs also to target net standard.

And is it a good idea to use multi-target to solve this temporarily?

**Comment by 333fred** [comment]
@jmarolf and @chsienki, can you give acaly advice here?

**Comment by chsienki** [comment, marked as answer]
@acaly  The generator project needs to target `netstandard2.0` if you plan to run it inside Visual Studio (VS is built on .NET Framework). You can certainly use multi-targeting of the dependent projects if you need to reference them from your generator to achieve that.

Note that multi-targeting will require your projects to be built multiple times for each target, which may slow your build times. Consider also extracting only the core pieces the generator relies on into a separate `netstandard` assembly that both the generator and your `netcore` targeting projects can reference.

**Reply by acaly** [reply]
Thank you for the advice! 

> if you plan to run it inside Visual Studio (VS is built on .NET Framework)

Does that mean we can manually ```dotnet build``` the project with ```netcore``` generators? (I don't think a solution incompatible with VS is very attractive though.)
